vendors/swedbank-pay-sdk-php/src/SwedbankPay/Api/Service/ResourceFactory.php

```diff
@@ -0,0 +1,359 @@
+
+        foreach ($resourceNamespaces as $namespace) {
+            if ($this->findFileByNamespace($namespace)) {
+                return '\\' . preg_replace("/[\\\\]+/", '\\', $namespace);
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param $resource
+     * @param string $service
+     * @param string $type
+     * @return array
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
+    private function getResourceNsLookups($resource, $service = '', $type = '')
+    {
+        $resource = $this->camelCaseStr($resource);
+
+        if ($service) {
+            $service = implode('\\', array_map([$this, 'camelCaseStr'], explode('/', $service))) . '\\';
+        }
+
+        $type = ($type) ? $this->camelCaseStr($type) . '\\' : '';
+
+        $resourceNsLookUps = [];
+
+        switch ($type) {
+            case 'Collection\\':
+                $resourceNsLookUps[] = __NAMESPACE__ . "\\{$service}Resource\\Collection\\{$resource}Collection";
+                if ($service && $service != 'Payment\\') {
+                    $resourceNsLookUps[] = __NAMESPACE__ . "\\Payment\\Resource\\Collection\\{$resource}Collection";
+                }
+                $resourceNsLookUps[] = __NAMESPACE__ . "\\Resource\\Collection\\{$resource}Collection";
+                break;
+            default:
+                if ($type) {
+                    $resourceNsLookUps[] = __NAMESPACE__ . "\\{$service}Resource\\{$type}{$resource}";
+                    if ($service && $service != 'Payment\\') {
+                        $namespace = __NAMESPACE__ . "\\Payment\\Resource\\{$type}{$resource}";
+                        if (strpos($service, 'Transaction')) {
+                            $namespace = __NAMESPACE__ . "\\Payment\\Transaction\\Resource\\{$type}{$resource}";
+                        }
+
+                        $resourceNsLookUps[] = $namespace;
+                    }
+                    $resourceNsLookUps[] = __NAMESPACE__ . "\\Resource\\{$type}{$resource}";
+                }
+                $resourceNsLookUps[] = __NAMESPACE__ . "\\{$service}Resource\\{$resource}";
+                if ($service && $service != 'Payment\\') {
+                    $namespace = __NAMESPACE__ . "\\Payment\\Resource\\{$resource}";
+                    if (strpos($service, 'Transaction')) {
+                        $namespace = __NAMESPACE__ . "\\Payment\\Transaction\\Resource\\{$resource}";
+                    }
+
+                    $resourceNsLookUps[] = $namespace;
+                }
+                $resourceNsLookUps[] = __NAMESPACE__ . "\\Resource\\{$resource}";
+                break;
+        }
+
+        return $resourceNsLookUps;
+    }
+
+    private function findFileByNamespace($resourceFqcn)
+    {
+        $basePath = str_replace(str_replace('\\', '/', __NAMESPACE__), '', __DIR__);
+        return file_exists($basePath . str_replace('\\', '/', $resourceFqcn) . '.php');
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function file_exists() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-checkout/pullRequest?prid=5510424)